source/command-monitoring/tests/unified/redacted-commands.yml

```diff
@@ -0,0 +1,166 @@
+description: "redacted-commands"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "5.0"
+
+createEntities:
+  - client:
+      id: &client client
+      observeEvents:
+        - commandStartedEvent
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName command-monitoring-tests
+
+tests:
+  - description: "authenticate"
+    operations:
+      - name: runCommand
+        object: *database
+        arguments:
+          commandName: authenticate
+          command:
+            authenticate: "private"
+        # Malformed authentication commands will fail against the server, but we
+        # just want to assert that security-related commands are redacted
+        # in command monitoring.
+        expectError:
```

Comment:
I thought it would be easier to "fake" most of these authentication related commands and just expect the server error since most of them are malformed (missing fields like `mechanism`, "private" is an invalid value, or the command cannot be run out of order as in `saslContinue`).

I _could_ make sure each command is run with the correct context, so they all can succeed, but we're not really testing authentication in drivers here, just command monitoring redaction.

Reply:
I agree.